### DISCUSSION TITLE
Accessing internal variables in Dialogic 2

### DISCUSSION THREAD

**Discussion by Kozelek** [dialogic-godot/dialogic#1556] [Q&A] [answered]
I saw that in Dialogic 1 it was possible to access internal variables with the Dialogic Class and functions get_variable(name, default) and set_variable(name, default).
Is it possible to do the same in Dialogic 2?
I need it because I want to make a "map selection" scene where the player can select several places from a city. Once a place in the city is selected it should not be selected again, that's why I need to control that with Dialogic's internal variables.

**Comment by Jowan-Spooner** [comment, marked as answer]
In dialogic 2 you can access variables via the Variable subsystem, shortened to VAR:

If your dialogic-setup looks like this:
![grafik](https://github.com/coppolaemilio/dialogic/assets/42868150/2754148d-d779-4990-b837-8f0d4fbf4f42)

You could write the following in code:
```gdscript
Dialogic.VAR.Player.Name = "Emily"
if Dialogic.VAR.Counter > 3:
    Dialogic.VAR.Player.Health += 20
```
*This is also explained in more detail in the variables editor.*

---
Alternatively you can now access external variables of Autoloads in conditions. So if you had for example a "Map" autoload with variables that indicate which places have been visited you could make the following conditions:
![grafik](https://github.com/coppolaemilio/dialogic/assets/42868150/bb71e5db-4df2-4d13-9b00-2f26dcf01d32)

That would imply this kind of variable in your "Map" autoloads script:
``` gdscript
var visited_places := ["Kitchen", "School", "Park"]
```

**Comment by Kozelek** [comment]
Thanks! This definitely helps!
